### PR TITLE
Test runners using ApplicationMessage.Log

### PR DESCRIPTION
**Pull request by nnikolopoulos**

### CODE REVIEW

**Comment by nnikolopoulos** [comment]
It turns out that the ImportStorageTest had some failing test that were suppressed and are now visible thanks to the ActivityLog and the fluent API

image

Not sure how to fix the failing test.

File: ifrs17-template/Test/ReportStorageTest.ipynb [hunk 1]
```diff
@@ -77,14 +79,36 @@
         {
             "cell_type": "code",
             "source": [
-                "var testData = new ExchangeRate[] {new ExchangeRate{ Currency = \"EUR\", Year = 2020, Month = 12, FxType = FxType.Spot, FxToGroupCurrency = 10 }, ",
+                "using System.ComponentModel.DataAnnotations;",
```
Comment (sbusato) [inline ifrs17-template/Test/ReportStorageTest.ipynb:82]: This line should not be needed.

File: ifrs17-template/Test/ReportStorageTest.ipynb [hunk 2]
```diff
@@ -77,14 +79,36 @@
...
                 "\n                                 new ExchangeRate{ Currency = \"EUR\", Year = 2021, Month = 6, FxType = FxType.Average, FxToGroupCurrency = 20 },",
                 "\n                                 new ExchangeRate{ Currency = \"EUR\", Year = 2021, Month = 6, FxType = FxType.Spot, FxToGroupCurrency = 30 },",
                 "\n                                 new ExchangeRate{ Currency = \"USD\", Year = 2020, Month = 12, FxType = FxType.Spot, FxToGroupCurrency = 5 },",
                 "\n                                 new ExchangeRate{ Currency = \"USD\", Year = 2021, Month = 6, FxType = FxType.Average, FxToGroupCurrency = 2 },",
                 "\n                                 new ExchangeRate{ Currency = \"USD\", Year = 2021, Month = 6, FxType = FxType.Spot, FxToGroupCurrency = 0.5 }};",
-                "\nawait CheckGetFx(\"EUR\", \"USD\", 2021, 6, testData, 2, 10, 60);"
+                "\nvar testDataResults = new List<ValidationResult>();",
+                "\nforeach (var instance in testData)  Validator.TryValidateObject(instance, new ValidationContext(instance), testDataResults, true);",
+                "\ntestDataResults"
```
Comment (sbusato) [inline ifrs17-template/Test/ReportStorageTest.ipynb:92]: Clean up the lines above.

File: ifrs17/Test/QueriesTest.ipynb [hunk 3]
```diff
@@ -190,11 +282,33 @@
...
             "source": [
-                "var args = new Args(\"EUR\",2020,9,Periodicity.Monthly,scenario);",
+                "using System.ComponentModel.DataAnnotations;",
+                "\nvar args = new Args(\"EUR\",2020,9,Periodicity.Monthly,scenario);",
                 "\nvar testData = new YieldCurve[] {new YieldCurve{ Currency = \"EUR\", Year = 2020, Month = 9, Scenario = scenario},",
                 "\n                                 new YieldCurve{ Currency = \"EUR\", Year = 2019, Month = 12, Scenario = scenario},",
                 "\n                                 new YieldCurve{ Currency = \"EUR\", Year = 2019, Month = 12, Scenario = null}};",
-                "\nawait CheckLoadCurrentAndPreviousParameterForYieldCurveAsync(args, testData, (2020, 9, scenario), (2019, 12, null));"
+                "\nvar activity = await CheckLoadCurrentAndPreviousParameterForYieldCurveAsync(args, testData, (2020, 9, scenario), (2019, 12, null));",
+                "\nvar testDataResults = new List<ValidationResult>();",
```
Comment (dcolleoni) [inline ifrs17/Test/QueriesTest.ipynb:291]: Remove all this Validator checks

**Review by dcolleoni — APPROVED** [review]

**Pull request merged**